### PR TITLE
fp: Refactor skolem handling of FpExpandDefs.

### PR DESCRIPTION
**Pull request by aniemetz**
This refactors partial function handling to use the skolem infrastructure the right way (use mkSkolemFunction instead of mkDummySkolem).

Fixes cvc5/cvc5-projects#265.

### CODE REVIEW

File: include/cvc5/cvc5_skolem_id.h [hunk 1]
```diff
@@ -654,6 +654,33 @@ enum ENUM(SkolemId) : uint32_t
    * - Type: ``E``
    */
   EVALUE(SETS_MAP_DOWN_ELEMENT),
+  /**
+   * A skolem function that is unique per floating-point sort, introduced for
```
Comment (ajreynol) [inline include/cvc5/cvc5_skolem_id.h:658]: Can you document the skolem indices and type of each of these (see the style of the other enum values).

**Review by ajreynol — CHANGES_REQUESTED** [review]

File: include/cvc5/cvc5_skolem_id.h [hunk 2]
```diff
@@ -654,6 +654,50 @@ enum ENUM(SkolemId) : uint32_t
...
+   * the undefined zero case of ``fp.min``.
+   *
+   * - Number of skolem indices: ``1``
+   *   - ``1:`` The Sort of the fp.min operator.
```
Comment (ajreynol) [inline include/cvc5/cvc5_skolem_id.h:662]: Document the types of the skolems as well.

Comment (aniemetz) [reply include/cvc5/cvc5_skolem_id.h:662]: Why do we document this as `Type` when this is also visible on the API level?

File: include/cvc5/cvc5_skolem_id.h [hunk 3]
```diff
@@ -651,9 +651,58 @@ enum ENUM(SkolemId) : uint32_t
...
+   * - Sort: ``E``
    */
   EVALUE(SETS_MAP_DOWN_ELEMENT),
+  /**
+   * A skolem function that is unique per floating-point sort, introduced for
+   * the undefined zero case of ``fp.min``.
+   *
+   * - Number of skolem indices: ``1``
+   *   - ``1:`` The floating-point sort ``FP`` of the fp.min operator.
+   * - Sort: ``(-> (FP FP) (_ BitVector 1)``
```
Comment (ajreynol) [inline include/cvc5/cvc5_skolem_id.h:663]:
```suggestion
   * - Sort: ``(-> FP FP (_ BitVector 1)``
```

File: include/cvc5/cvc5_skolem_id.h [hunk 4]
```diff
@@ -651,9 +651,58 @@ enum ENUM(SkolemId) : uint32_t
...
+   */
+  EVALUE(FP_MIN_ZERO),
+  /**
+   * A skolem function that is unique per floating-point sort, introduced for
+   * the undefined zero case of ``fp.max``.
+   *
+   * - Number of skolem indices: ``1``
+   *   - ``1:`` The floating-point sort ``FP`` of the fp.max operator.
+   * - Sort: ``(-> (FP FP) (_ BitVector 1)``
```
Comment (ajreynol) [inline include/cvc5/cvc5_skolem_id.h:672]:
```suggestion
   * - Sort: ``(-> FP FP (_ BitVector 1)``
```

File: include/cvc5/cvc5_skolem_id.h [hunk 5]
```diff
@@ -651,9 +651,58 @@ enum ENUM(SkolemId) : uint32_t
...
+  EVALUE(FP_MAX_ZERO),
+  /**
+   * A skolem function introduced for the undefined out-ouf-bounds case of
+   * ``fp.to_ubv`` that is unique per floating-point sort and sort of the
+   * arguments to the operator.
+   *
+   * - Number of skolem indices: ``2``
+   *   - ``1:`` The floating-point sort ``FP`` of operand of fp.to_ubv.
+   *   - ``2:`` The bit-vector sort ``BV`` to convert to.
+   * - Sort: ``(-> (RoundingMode FP) BV)``
```
Comment (ajreynol) [inline include/cvc5/cvc5_skolem_id.h:683]:
```suggestion
   * - Sort: ``(-> RoundingMode FP BV)``
```

File: include/cvc5/cvc5_skolem_id.h [hunk 6]
```diff
@@ -651,9 +651,58 @@ enum ENUM(SkolemId) : uint32_t
...
+  EVALUE(FP_TO_UBV),
+  /**
+   * A skolem function introduced for the undefined out-ouf-bounds case of
+   * ``fp.to_sbv`` that is unique per floating-point sort and sort of the
+   * arguments to the operator.
+   *
+   * - Number of skolem indices: ``2``
+   *   - ``1:`` The floating-point sort ``FP`` of operand of fp.to_sbv.
+   *   - ``2:`` The bit-vector sort ``BV`` to convert to.
+   * - Sort: ``(-> (RoundingMode FP) BV)``
```
Comment (ajreynol) [inline include/cvc5/cvc5_skolem_id.h:694]:
```suggestion
   * - Sort: ``(-> RoundingMode FP BV)``
```

**Review by ajreynol — APPROVED** [review]
LGTM, thanks for the updates.  Just minor fixes, the syntax for function sorts is n-ary (no nested parentheses).

**Pull request merged**